workbench_access_protect/src/Access/DeleteAccessCheck.php

```diff
@@ -0,0 +1,263 @@
+      $return = FALSE;
+    }
+
+    return $return;
+  }
+
+  /**
+   * @{inheritdoc}
+   */
+  public function getBundles(EntityInterface $entity) {
+    $bundle = $this->entityTypeManager->getDefinitions()[$entity->getEntityTypeId()]->get('bundle_of');
+    $bundles = $this->entityTypeBundleInfo->getBundleInfo($bundle);
+
+    return array_combine(array_keys($bundles), array_keys($bundles));
+  }
+
+  /**
+   * @{inheritdoc}
+   */
+  public function hasBundles(EntityInterface $entity) {
+    if ($this->entityTypeManager->getDefinitions()[$entity->getEntityTypeId()]->get('bundle_of') == null) {
+      return FALSE;
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isDeleteAllowedBundle($bundle, $entity){
+    $bundle_of = $this->entityTypeManager->getDefinitions()[$entity->getEntityTypeId()]->get('bundle_of');
+    $entity_id_key = $this->entityTypeManager->getDefinitions()[$entity->getEntityTypeId()]->get('entity_keys')['id'];
+    $entities = $this->entityTypeManager->getStorage($bundle_of)->loadByProperties(
+      [$entity_id_key => $bundle ]
+    );
+
+    /*
+     * Cycle through the entities of this bundle. As soon as one is discovered
+     * as being actively used for access control, we can deny delete.
+     */
+    foreach ($entities as $bundle) {
+      if ($this->isDeleteAllowed($bundle) === FALSE) {
+        return FALSE;
+      }
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Determines if this term has active members in it.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to inspect.
+   *
+   * @return bool
+   *   TRUE if the term has members, FALSE otherwise.
+   */
+  private function hasMembers(EntityInterface $entity) {
+    return (bool) (count($this->getActiveSections($entity)) > 0);
+  }
+
+  /**
+   * Inspects the given entity for active users.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to inspect.
+   *
+   * @return array
+   *   An array of the users assigned to this section.
+   *
+   */
+  private function getActiveSections(EntityInterface $entity) {
+    /** @var \Drupal\workbench_access\UserSectionStorageInterface $sectionStorage */
+    $sectionStorage = $this->userSectionStorage;
+
+    $editors = array_reduce($this->entityTypeManager->getStorage('access_scheme')->loadMultiple(),
+      function (array $editors, AccessSchemeInterface $scheme) use ($sectionStorage, $entity) {
+      $editors += $sectionStorage->getEditors($scheme, $entity->id());
+      return $editors;
+    }, []);
+
+    return $editors;
+  }
+
+  /**
+   * Inspects the given entity for active content.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to inspect.
+   *
+   * @return bool
+   *   TRUE if content is assigned to this entity.
+   *   FALSE if content is not assigned to this entity.
+   */
+  private function hasContent(EntityInterface $entity) {
```

Comment:
add a call to a method here that will check to see if the term has children or not.
TODO: figure out how to abstract this so that this class doesn't have any explicit taxonomy or term references. Adding a method to the Drupal\workbench_access\Plugin\AccessControlHierarchy class to see if this scheme has children might be the way to do it.